llvm/lib/IR/Globals.cpp

```diff
@@ -78,6 +78,46 @@ GlobalValue::getGUIDAssumingExternalLinkage(StringRef GlobalIdentifier) {
   return MD5Hash(GlobalIdentifier);
 }
 
+void GlobalValue::assignGUID() {
```

Comment:
does this need to be on `GlobalValue`? It can just be a detail of the implementation of `AssignGUIDPass`, and then there's no confusion on the user side about whether it's something they should call or not.

Reply:
I wonder if it would be better to make AssignGUIDPass::run a friend so that other users cannot access this functionality.